host_camera_service/host_camera_server.cc

```diff
@@ -447,12 +442,13 @@ this_thread::sleep_for(32ms);
              << ex.what() << endl;
         exit(1);
     }
-        int err = pthread_create(&t_handle, NULL, &InitCamera, NULL);
-if(err != 0)
-	cout << "Fail to create thread\n";
+    int err = pthread_create(&t_handle, NULL, &InitCamera, NULL);
+    if(err != 0)
+        cout << "Fail to create thread\n";
     // we need to be alive :)
     while (true) {
-        this_thread::sleep_for(33ms);
+	pthread_mutex_lock(&mMainLock);
+        pthread_cond_wait(&mSignalMain, &mMainLock);
```

Comment:
I think, Ideally this mSignalMain should be signalled based on some std::signal event.
Here, I guess this is never signaled. 
Either case, I think you can get rid of the while loop, since this would indefinitely wait for the signal.